### PR TITLE
TweakReg regression test fix.

### PR DESCRIPTION
**Pull request by mairanteodoro**
This PR implements the following items to fix TweakReg's (TR) regression test:

- fix TR's catalog metadata: a bug was preventing the catalog metadata `name` from being properly set;
- updated the files used by TweakReg's regression test:
   * input: `roman-pipeline/dev/WFI/image/r0000501001001001001_01101_0001_WFI02_cal_tweakreg.asdf`;
   * truth: `roman-pipeline/dev/truth/WFI/image/r0000501001001001001_01101_0001_WFI02_tweakreg.asdf`.

The truth file in Artifactory was created by processing the cal file through the `TweakRegStep`:
```python
from romancal.stpipe import RomanStep
from romancal.tweakreg.tweakreg_step import TweakRegStep

step = TweakRegStep()

args = [
    "romancal.step.TweakRegStep",
    "/full/path/to/association/file/asn_filename.json",
    "--output_file='tweakreg_truth_filename.asdf'",
    "--suffix='output'",
]
RomanStep.from_cmdline(args)
```
The association file (named `asn_filename.json` above) used in the creation of the truth file has the following content:
```json
{
       "asn_type": "None",
       "asn_rule": "DMS_ELPP_Base",
       "version_id": null,
       "code_version": "0.9.1.dev28+ge987cc9.d20230106",
       "degraded_status": "No known degraded exposures in association.",
       "program": "noprogram",
       "constraints": "No constraints",
       "asn_id": "a3001",
       "target": "none",
       "asn_pool": "test_pool_name",
       "products": [
           {
               "name": "files.asdf",
               "members": [
                   {
                       "expname": "r0000501001001001001_01101_0001_WFI02_cal_tweakreg.asdf",
                       "exptype": "science"
                   }
               ]
            }
       ]
}
```

The file used as input for the regression test, `r0000501001001001001_01101_0001_WFI02_cal.asdf`, was provided by @tddesjardins and has been processed through `assign_wcs`, `photom`, and `source_detection`. We added the `tweakreg` suffix before uploading it to Artifactory.

**Astrometric Utils unit tests**
This PR also adds parallax correction to the unit tests for the `astrometric_utils` package.

Since the coordinates in the catalog obtained from the VO API are now being updated for proper motion and parallax, we must consider that in the `test_astrometric_utils` suite.

**Regression test**
~All regression tests are passing: https://plwishmaster.stsci.edu:8081/job/RT/job/Roman-Developers-Pull-Requests/401/~
${\text{\color{red}The regression tests are failing to start and Zach and William are working on fixing it.}}$

**Checklist**
- [ ] added entry in `CHANGES.rst` under the corresponding subsection
- [X] updated relevant tests
- [ ] updated relevant documentation
- [ ] updated relevant milestone(s)
- [ ] added relevant label(s)
- [X] ran regression tests, post a link to the Jenkins job below. [How to run regression tests on a PR](https://github.com/spacetelescope/romancal/wiki/Running-Regression-Tests-Against-PR-Branches)

### CODE REVIEW

**Comment by ddavis-stsci** [comment]
Is there some reason you are switching from WFI01 to WFI02? 
Does this mean that we need to retire the files for the older tests once this is in? 
The WFI02 file is also a custom file from INS (Tyler)?

**Comment by mairanteodoro** [comment]
> Is there some reason you are switching from WFI01 to WFI02?

Yes; the reason is that seven files are throwing an error (`ValueError: Out of domain for acos`) when calculating the area of the spherical triangles during the `TweakRegStep` processing. The files are:

- `r0000501001001001001_01101_0001_WFI07_cal.asdf`;
- `r0000501001001001001_01101_0001_WFI16_cal.asdf`;
- `r0000501001001001001_01101_0001_WFI17_cal.asdf`;
- `r0000501001001001001_01101_0002_WFI01_cal.asdf`;
- `r0000501001001001001_01101_0002_WFI03_cal.asdf`;
- `r0000501001001001001_01101_0002_WFI09_cal.asdf`;
- `r0000501001001001001_01101_0002_WFI10_cal.asdf`.

If we can't process the files through `TweakRegStep`, we can't pass it to `ResampleStep`. Both @tddesjardins and @nden are aware of this issue.

For the resample regression test specifically, we are using SCA 02 because that allows us to test the alignment of the same SCA but from different exposures (the latest simulations have a small offset between exposures). SCA 01 can't be used for that since it can't be processed through `TweakRegStep`.


> Does this mean that we need to retire the files for the older tests once this is in?

I don't think so; this issue affects only TweakReg.

> The WFI02 file is also a custom file from INS (Tyler)?

Yes.

**Comment by ddavis-stsci** [comment]
Do we have a timescale of when the "error (ValueError: Out of domain for acos)"  might be fixed? This clearly cannot go into the elpp until this is working. 
At that point we should revert the file to WFI01, or convert all the tests to use WFI02.

**Comment by mairanteodoro** [comment]
> Do we have a timescale of when the "error (ValueError: Out of domain for acos)" might be fixed? This clearly cannot go into the elpp until this is working. At that point we should revert the file to WFI01, or convert all the tests to use WFI02.

The ticket for that work is https://jira.stsci.edu/browse/RCAL-688.

File: romancal/tweakreg/tests/test_astrometric_utils.py [hunk 1]
```diff
@@ -481,39 +598,51 @@ def test_get_catalog_valid_parameters_but_no_sources_returned():
...
     )
 
     assert len(result) > 0
-    assert np.isclose(returned_ra, expected_new_ra, atol=1e-10, rtol=1e-9).all()
-    assert np.isclose(returned_dec, expected_new_dec, atol=1e-10, rtol=1e-9).all()
+    # choosing atol=0 corresponds to abs(a - b) / abs(b) <= rtol,
+    # where a is the returned value from the VO API and b is the expected
+    # value from applying the calculated PM and parallax corrections.
+    assert np.isclose(returned_ra, expected_ra, atol=0, rtol=1e-5).all()
+    assert np.isclose(returned_dec, expected_dec, atol=0, rtol=1e-5).all()
```
Comment (schlafly) [inline romancal/tweakreg/tests/test_astrometric_utils.py:645]: An rtol of 1e-5 worries me?  For an object with a coordinate of something like 100, this is 1e-3 deg = 3.6" = 36 pixels?  Even for something with a coordinate of ~1, this is 0.36 pixels while we want to be doing more like 0.01 pixel.  Maybe just an atol of 0.001 and no rtol?

Comment (mairanteodoro) [reply romancal/tweakreg/tests/test_astrometric_utils.py]: Changed `atol` to `1e-5` and `rtol=0`.

**Comment by schlafly** [comment]
What is the actual level of agreement you're getting?

I said atol = 0.001 but I'm not sure what I was thinking; that's the 3.6" agreement we were talking about before.  You did atol=1e-5, which is 0.036" = 0.36 pixel, which is not great; we really want ~0.01 - 0.001 pix or something.

**Comment by mairanteodoro** [comment]
> What is the actual level of agreement you're getting?
> 
> I said atol = 0.001 but I'm not sure what I was thinking; that's the 3.6" agreement we were talking about before. You did atol=1e-5, which is 0.036" = 0.36 pixel, which is not great; we really want ~0.01 - 0.001 pix or something.

We are using `atol=1e-8` (absolute coordinates difference, in deg) in the unit tests, which corresponds to an absolute difference in coordinates of $<$ 40 &mu;as.

**Comment by mairanteodoro** [comment]
The tolerance for coordinates correction has been set to $2.8\times10\^{-9}$ deg, which corresponds to 10 $\mu$ as

**Review by schlafly — APPROVED** [review]
Thanks Mairan.  That looks good to me.

**Pull request merged**